src/main/scala/com/amazon/deequ/anomalydetection/seasonal/HoltWinters.scala

```diff
@@ -0,0 +1,233 @@
+  * express or implied. See the License for the specific language governing
+  * permissions and limitations under the License.
+  *
+  */
+
+package com.amazon.deequ.anomalydetection.seasonal
+
+import breeze.linalg.DenseVector
+import breeze.optimize.{ApproximateGradientFunction, DiffFunction, LBFGSB}
+import com.amazon.deequ.anomalydetection.{Anomaly, AnomalyDetectionStrategy}
+
+import collection.mutable.ListBuffer
+
+object HoltWinters {
+
+  object SeriesSeasonality extends Enumeration {
+    val Weekly, Yearly: Value = Value
+  }
+
+  object MetricInterval extends Enumeration {
+    val Daily, Monthly: Value = Value
+  }
+
+  private[seasonal] case class ModelResults(
+    forecasts: Seq[Double],
+    level: Seq[Double],
+    trend: Seq[Double],
+    seasonality: Seq[Double],
+    residuals: Seq[Double]
+  )
+
+}
+
+/**
+  * Detects anomalies based on additive Holt-Winters model. The methods has two
+  * parameters, one for the expected periodicity and one for the metric frequency,
+  * as in how often the metric of interest is computed (e.g. daily). The periodicity
+  * is the longest cycle in the data to be expected.
+  *
+  * @param seasonality: Cycle length of the metric
+  * @param metricsInterval: How often a metric is available
+  */
+class HoltWinters(
+    seasonality: HoltWinters.SeriesSeasonality.Value,
+    metricsInterval: HoltWinters.MetricInterval.Value)
+  extends AnomalyDetectionStrategy {
+
+  import HoltWinters._
+
+  private val seriesPeriodicity = seasonality -> metricsInterval match {
+    case (SeriesSeasonality.Weekly, MetricInterval.Daily) => 7
+    case (SeriesSeasonality.Yearly, MetricInterval.Monthly) => 12
+  }
+
+  /**
+    * Triple exponential smoothing with additive trend and seasonality
+    * ETS(A, A) according to https://otexts.org/fpp2/taxonomy.html
+    *
+    * References:
+    * - https://otexts.org/fpp2/holt-winters.html
+    * - https://gist.github.com/andrequeiroz/5888967
+    *
+    * @param series: input time series
+    * @param periodicity: periodicity of series
+    * @param numberOfPointsToForecast: number of data points to forecast
+    * @return (forecasts, one step ahead residual SD)
+    */
+  private[seasonal] def additiveHoltWinters(
+      series: Seq[Double],
+      periodicity: Int,
+      numberOfPointsToForecast: Int,
+      alpha: Double,
+      beta: Double,
+      gamma: Double)
+    : ModelResults = {
+
+    val firstPeriodSum = series.take(periodicity).sum
+    val secondPeriodSum = series.slice(periodicity, 2 * periodicity).sum
+
+    // mean of first period
+    val level = ListBuffer(firstPeriodSum / periodicity.toDouble)
+
+    // mean of second period - mean of first period
+    val trend = ListBuffer(
+      (secondPeriodSum - firstPeriodSum) / (periodicity.toDouble * periodicity.toDouble)
+    )
+
+    // first `periodicity` data points - estimated level
+    val seasonality = ListBuffer(series.take(periodicity).map(_ - level.head): _*)
```

Comment:
move this code out of the constructor for the listbuffer, its really hard to read with the : _* operation at the end